@shared/api/internal.ts

```diff
@@ -1221,6 +1221,10 @@ export const saveSettings = async ({
     console.error(e);
   }
 
+  if (response.error) {
+    throw new Error(response.error);
+  }
```

Comment:
now that an account mismatch returns an error, we should make sure all of our services properly handle errors